src/gradualizer_fmt.erl

```diff
@@ -46,12 +46,17 @@ format_type_error({type_error, Expression, ActualType, ExpectedType}, Opts)
   when is_tuple(Expression) ->
     format_expr_type_error(Expression, ActualType, ExpectedType, Opts);
 format_type_error({nonexhaustive, Anno, Example}, Opts) ->
+    FormattedExample =
+        case io_lib:deep_char_list(Example) of
+            true -> ["\n", Example];
+            false -> io_lib:format("~p", [Example])
+        end,
```

Comment:
The `["\n", Example]` is to align the printed record to the leftmost column.

It makes more sense to have this logic here (instead of pick_value), but the downside is that an exhaustive check for a list could return something like `[40]` which would be showed on a newline.

Reply:
Printing the example on a separate line should rather depend on if it's multiline or not. It could be multiline also if it's large map, tuple, etc. and a record could fit on a single line if it's small.

Besides, I don't like that this can be either already pretty-printed stuff or any Erlang value for two reasons: ambiguity and complexity.